osu.Server.Queues.ScoreStatisticsProcessor/Stores/BeatmapStore.cs

```diff
@@ -107,12 +131,40 @@ public static async Task<BeatmapStore> CreateAsync(MySqlConnection connection, M
             return difficultyAttributes;
         }
 
+        /// <remarks>
+        /// This method attempts to create a simple solution to deciding if a <see cref="Mod"/> can be considered a ranked "legacy" mod.
+        /// Used by <see cref="GetDifficultyAttributesAsync"/> to decide if the current mod combination's difficulty attributes
+        /// can be fetched from the database.
+        /// </remarks>
+        private static bool isRankedLegacyMod(Mod mod) =>
+            mod is ModNoFail
+                or ModEasy
+                or ModHidden // this also catches ManiaModFadeIn
```

Comment:
> Also I'd probably slightly prefer if this was listing leaf classes in the inheritance classes (don't like the // this also catches ManiaModFadeIn comment, and worried about future accidental inheritance)

Case in point: this will _also_ catch `ManiaModCover`, [because `ManiaModCover : ManiaModWithPlayfieldCover : ModHidden`](https://github.com/ppy/osu/blob/7e5fda45396ce7823a7b5e63c7204c9ceafd184c/osu.Game.Rulesets.Mania/Mods/ManiaModCover.cs#L13).

Reply:
Should be solved in https://github.com/ppy/osu-queue-score-statistics/pull/274/commits/09efc446d830650e0d28142b1c394058c64cfade.